core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,240 @@
+        return Mono.just(pollResponse);
+    }
+
+    /**
+     * This will block till poll operation is complete
+     * @return returns poll response
+     */
+    public PollResponse<T> block() {
+        return getObserver().blockLast();
+    }
+
+    /*
+     * Calls poll operation function and update pollResponse.
+     */
+    private  void updatePollOperationSync() {
+        pollResponse = pollOperation.apply(pollResponse);
+    }
+
+    /*
+     * Get whether or not this PollStrategy's long running operation is done.
+     * @return Whether or not this PollStrategy's long running operation is done.
+     */
+    private Mono<PollResponse<T>> sendPollRequestWithDelay() {
+        return Mono.defer(() -> delayAsync().then(Mono.defer(() -> {
+            if (!isTerminalState()) {
+                updatePollOperationSync();
+            } else if (!isTerminalState()) {
+                return Mono.empty();
+            }
+            return Mono.just(pollResponse);
+        })));
+    }
+
+    /*
+     * Helper function
+     */
+    private boolean isTerminalState() {
+        return pollResponse != null && pollResponse.isDone();
+
+    }
+
+    /*
+     * If this PollerOptions has a pollIntervalInMillis value, return an Mono that is delayed by the
+     * number of seconds that are in the pollIntervalInMillis value. If this PollerOptions doesn't have
+     * a pollIntervalInMillis value, then return an Single with no delay.
+     *
+     * @return A Mono with delay if this PollerOptions has a pollIntervalInMillis value.
+     */
+    private Mono<Void> delayAsync() {
+        Mono<Void> result = Mono.empty();
+        if (this.pollerOptions.pollInterval().toNanos() > 0) {
+            result = result.delaySubscription(this.pollerOptions.pollInterval());
+        }
+        return result;
+    }
+
+    /**
+     *
+     * @param  autoPollingEnabled  true : Ensures the polling is happening in background.
+     *                             false : Ensures that polling is not happening in background.
+     */
+    public void setAutoPollingEnabled(boolean autoPollingEnabled) {
+    	
+    	// setting same auto polling status would not require any action.
+    	
+    	if (this.autoPollingEnabled == autoPollingEnabled ) {
+    		return;
+    	}
+
+    	this.autoPollingEnabled = autoPollingEnabled;
+        if (this.autoPollingEnabled) {
+            if (valid(fluxDisposable)) {
+                fluxDisposable = fluxHandle.subscribe(pr ->  pollResponse = pr );
+            }
+        } else {
+        	if (valid(fluxDisposable)) {
+                fluxDisposable.dispose();
+            }
+            fluxHandle = null;
+        }
+    }
+    
+    private boolean valid(Disposable disposable) {
+    	return (fluxDisposable != null && fluxDisposable.isDisposed());
+    		  
+    	 
+    	
+    }
+
+    /**
+     * @return true if polling is stopped.
+     */
+    public boolean isAutoPollingEnabled() {
+        return !this.autoPollingEnabled;
```

Comment:
return the opposite? 😮  Either the method name or variable needs better naming?

Reply:
yeah I missed that . Fixed this AM